zerver/models.py

```diff
@@ -164,6 +163,13 @@ class Realm(models.Model):
     COMMUNITY = 2
     org_type = models.PositiveSmallIntegerField(default=CORPORATE)  # type: int
 
+    # This value is also being used in static/js/settings_bots.bots_allowed_for_non_admins_values.
+    # On updating it here, update it there as well.
+    ALL = 1
+    NO_GENERIC_BOTS = 2
+    NONE = 3
+    bots_allowed_for_non_admins = models.PositiveSmallIntegerField(default=ALL)  # type: int
```

Comment:
I think we can probably rename this to just `Realm.bot_creation_policy`.  

So you'd check `realm.bot_creation_policy == BOT_CREATION_UNLIMITED` to see if one can do anything.